File: src/main/java/com/weeth/domain/file/presentation/FileResponseCode.java
package com.weeth.domain.file.presentation;

import com.weeth.global.common.response.ResponseCodeInterface;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public enum FileResponseCode implements ResponseCodeInterface {

    PRESIGNED_URL_GET_SUCCESS(1500, HttpStatus.OK, "Presigned Url 반환에 성공했습니다");

    private final int code;
    private final HttpStatus status;
    private final String message;

    FileResponseCode(int code, HttpStatus status, String message) {
        this.code = code;
        this.status = status;
        this.message = message;
    }

}

File: src/main/kotlin/com/weeth/domain/file/presentation/FileResponseCode.kt
package com.weeth.domain.file.presentation

import com.weeth.global.common.response.ResponseCodeInterface
import org.springframework.http.HttpStatus

enum class FileResponseCode(
    override val code: Int,
    override val status: HttpStatus,
    override val message: String,
) : ResponseCodeInterface {
    PRESIGNED_URL_GET_SUCCESS(1500, HttpStatus.OK, "Presigned Url 반환에 성공했습니다"),
}
